src/Tasks/Common/src/ConflictResolution/FrameworkListReader.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.NET.Build.Tasks;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml.Linq;
+
+namespace Microsoft.NET.Build.Tasks.ConflictResolution
+{
+    static class FrameworkListReader
+    {
+        public static IEnumerable<ConflictItem> LoadConflictItems(string frameworkListPath, ILog log)
+        {
+            if (frameworkListPath == null)
+            {
+                throw new ArgumentNullException(nameof(frameworkListPath));
+            }
+
+            if (!File.Exists(frameworkListPath))
+            {
+                //  This is not an error, as we get both the root target framework directory as well as the Facades folder passed in as TargetFrameworkDirectories.
+                //  Only the root will have a RedistList\FrameworkList.xml in it
+                yield break;
+            }
+
+            var frameworkList = XDocument.Load(frameworkListPath);
+            foreach (var file in frameworkList.Elements("File"))
```

Comment:
Some framework lists chain to others. Critically, I believe Mono's are that way and this will need to work when targeting net471 with Mono msbuild.

Reply:
I took a look at this and it turns out that the `GetReferenceAssemblyPaths` target takes care of this.  If there are chained framework lists, it will set `TargetFrameworkDirectory` to a semicolon-separated list of the chained folders.  For example:

> C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v6.0\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v5.1\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v5.0\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.4.87\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.4\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.3\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.2\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.1\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v4.0.3\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v2.3\;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v1.0\;;;C:\Program Files %28x86%29\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\MonoAndroid\v1.0\Facades\

However, the `FrameworkList.xml` files for Mono don't list any `File` entries.  So I think we will need to scan the directory for assemblies to treat as platform assemblies if we don't find any `File` items listed in the `FrameworkList.xml`.

Reply:
That could get expensive, if you do need to do that make sure you don't crack the assemblies at that point but instead create lazy items that point to the TargetFrameworkDirectory location.

I don't suppose MSBuild has an API built on top of their framework list / metadata cache?

Reply:
@ericstj Yes, that's the way I would do it.  I'm not sure at this point whether we need this to apply to Xamarin/Mono.

The MSBuild API (the `RedistList` class) isn't public and doesn't have any way to get a list of the framework assemblies, only to check if a given assembly name is part of the framework.  We could make an API for this public if we have to, but so far it doesn't look like we do.

Reply:
@akoeplinger @DustinCampbell